### PR TITLE
424 implicitdebug constraints in implementation

### PR DESCRIPTION
**Pull request by lorenzogentile404**
https://www.notion.so/veridise/Implicit-debug-constraints-in-implementation-53a84f3c2e4642f1b0cd49602d56cb49

### CODE REVIEW

File: hub/constraints/instruction-handling/call/standpoint.lisp [hunk 1]
```diff
@@ -35,8 +35,8 @@
                                   )
                   )
 
-(defconstraint   call-instruction---backward-setting-CREATE-instruction   ()
-                 (if-not-zero    PEEK_AT_SCENARIO
+(defconstraint   call-instruction---backward-setting-CALL-instruction   ()
+                 (debug (if-not-zero    PEEK_AT_SCENARIO
```
Comment (OlivierBBB) [inline hub/constraints/instruction-handling/call/standpoint.lisp:39]: OK

Comment (OlivierBBB) [inline hub/constraints/instruction-handling/call/standpoint.lisp:39]: OK for the name change

File: hub/columns/shared.lisp [hunk 2]
```diff
@@ -29,7 +29,7 @@
     ( CALLER_CONTEXT_NUMBER                     :i32 )
 
     ;;
-    ( CONTEXT_WILL_REVERT                       :binary@prove )
+    ( CONTEXT_WILL_REVERT                       :binary )
```
Comment (OlivierBBB) [inline hub/columns/shared.lisp:32]: Indeed not required.

**Review by OlivierBBB — APPROVED** [review]
We should revert the `(debug ...` constraints for now.

File: hub/constraints/generalities/context.lisp [hunk 3]
```diff
@@ -33,17 +33,14 @@
                     stack/HALT_FLAG))
 
 (defconstraint setting-CMC-and-XAHOY ()
-               (begin (is-binary                                                     CMC)
-                      (is-binary                                                   XAHOY)
```
[the remark this answers is not in the document]
Comment (lorenzogentile404) [reply hub/constraints/generalities/context.lisp:37]: Yes.

```
  ( CONTEXT_MAY_CHANGE                        :binary@prove )
    ( EXCEPTION_AHOY                            :binary@prove )
```

**Pull request merged**